gci/migrations/0001_initial.py

```diff
@@ -0,0 +1,101 @@
+                ('email_id', models.EmailField(max_length=254)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Organization',
+            fields=[
+                ('identifier', models.IntegerField(default=0, primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=100)),
+                ('logo', models.ImageField(blank=True, upload_to='')),
+                ('url', models.URLField(blank=True)),
+                ('summary', models.TextField(default='', max_length=300)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='ProgramYear',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('start_date', models.DateField()),
+                ('last_date', models.DateField()),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Student',
+            fields=[
+                ('identifier', models.IntegerField(default=0, primary_key=True, serialize=False)),
+                ('display_name', models.CharField(max_length=100)),
+                ('program_year', models.ManyToManyField(to='gci.ProgramYear')),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Tag',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=100)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Task',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=200)),
+                ('description', models.TextField(max_length=500)),
+                ('status', models.IntegerField(choices=[(1, 'Draft'), (2, 'Published')], default=1)),
+                ('max_instances', models.IntegerField(default=1)),
+                ('is_beginner', models.BooleanField(default=False)),
+                ('time_to_complete_in_days', models.IntegerField(default=3)),
+                ('external_url', models.URLField()),
+                ('metadata', models.TextField()),
+                ('last_modified', models.DateField()),
+                ('categories', models.ManyToManyField(to='gci.Category')),
+                ('mentors', models.ManyToManyField(to='gci.Mentor')),
+                ('org', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='gci.Organization')),
+                ('tags', models.ManyToManyField(to='gci.Tag')),
+            ],
+        ),
+        migrations.CreateModel(
+            name='TaskInstance',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
```

Comment:
why `auto_created=True` ?  this may be the best option...

Reply:
this is because you havent specified an identifier in the data model; a unique ID exist in the API reference.
It needs to be the primary key.